Casandra source Mapping tinyint to short(should be byte)

I am using  source connector to  push my data from cassandra   table to kafka topic but for the the column of data type tinyint it is try to convert it into short and i am getting excetption 

=> org.apache.kafka.connect.errors.ConnectException: Connector msg-connector error processing row Row[6181, 12084, 5341, NULL, NULL, NULL, {"cardType":"BASIC_CARD","cardId":null,"msgText":"uuqieybqibdi2w","msgContentType":"TEXT","msgAttachmentUrl":null,"caption":null}, 0, NULL, 0, Tue Apr 09 16:06:50 IST 2019, NULL, 1232433602057, NULL, 0, false, NULL, 3, NULL, 7, NULL, NULL, NULL, NULL, 1, 3, Tue Apr 09 16:06:50 IST 2019] for table chat_inbox.chat_message.
at com.datamountaineer.streamreactor.connect.cassandra.source.CassandraTableReader$$anonfun$process$1.applyOrElse(CassandraTableReader.scala:247)
at com.datamountaineer.streamreactor.connect.cassandra.source.CassandraTableReader$$anonfun$process$1.applyOrElse(CassandraTableReader.scala:215)
at scala.concurrent.Future$$anonfun$onSuccess$1.apply(Future.scala:119)
at scala.concurrent.Future$$anonfun$onSuccess$1.apply(Future.scala:117)
at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:36)
at scala.concurrent.impl.ExecutionContextImpl$AdaptedForkJoinTask.exec(ExecutionContextImpl.scala:121)
at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
Caused by: com.datastax.driver.core.exceptions.CodecNotFoundException: Codec not found for requested operation: [tinyint <-> java.lang.Short]


I checked yoursouce code and found this  line -> 
case DataType.Name.TINYINT | DataType.Name.SMALLINT => row.getShort(columnDef.getName)


expected => tinyint should be mapped to byte and small int to short

--- a/kafka-connect-cassandra/build.gradle
+++ b/kafka-connect-cassandra/build.gradle
@@ -17,7 +17,7 @@
 project(":kafka-connect-cassandra") {
 
     ext {
-        cassandraDriverVersion = "3.5.0"
+        cassandraDriverVersion = "3.7.1"
         cassandraVersion = "3.11.0"
         cassandraUnitVersion = "3.3.0.2"
         jPountzVersion = "1.3.0"

--- a/kafka-connect-cassandra/src/main/scala/com/datamountaineer/streamreactor/connect/cassandra/source/CassandraTypeConverter.scala
+++ b/kafka-connect-cassandra/src/main/scala/com/datamountaineer/streamreactor/connect/cassandra/source/CassandraTypeConverter.scala
@@ -38,9 +38,9 @@ class CassandraTypeConverter(private val codecRegistry: CodecRegistry,
                              private val setting: CassandraSourceSetting) extends StrictLogging {
 
   val mapper = new ObjectMapper()
-  val OPTIONAL_DATE_SCHEMA = org.apache.kafka.connect.data.Date.builder().optional().build()
-  val OPTIONAL_TIMESTAMP_SCHEMA = Timestamp.builder().optional().build()
-  val OPTIONAL_DECIMAL_SCHEMA = Decimal.builder(18).optional().build()
+  val OPTIONAL_DATE_SCHEMA: Schema = org.apache.kafka.connect.data.Date.builder().optional().build()
+  val OPTIONAL_TIMESTAMP_SCHEMA: Schema = Timestamp.builder().optional().build()
+  val OPTIONAL_DECIMAL_SCHEMA: Schema = Decimal.builder(18).optional().build()
 
   private val mappingCollectionToJson: Boolean = setting.mappingCollectionToJson
 
@@ -97,7 +97,8 @@ class CassandraTypeConverter(private val codecRegistry: CodecRegistry,
         //need to convert to string since the schema is set to String
         Option(row.getUUID(columnDef.getName)).map(_.toString).orNull
       case DataType.Name.BLOB => row.getBytes(columnDef.getName)
-      case DataType.Name.TINYINT | DataType.Name.SMALLINT => row.getShort(columnDef.getName)
+      case DataType.Name.SMALLINT => row.getShort(columnDef.getName)
+      case DataType.Name.TINYINT => row.getByte(columnDef.getName)
       case DataType.Name.INT => row.getInt(columnDef.getName)
       case DataType.Name.DOUBLE => row.getDouble(columnDef.getName)
       case DataType.Name.FLOAT => row.getFloat(columnDef.getName)

--- a/kafka-connect-cassandra/src/test/scala/com/datamountaineer/streamreactor/connect/cassandra/source/TestCassandraSourceTaskTimestamp.scala
+++ b/kafka-connect-cassandra/src/test/scala/com/datamountaineer/streamreactor/connect/cassandra/source/TestCassandraSourceTaskTimestamp.scala
@@ -43,7 +43,7 @@ class TestCassandraSourceTaskTimestamp extends WordSpec
   var tableName: String = _
 
   override def beforeAll {
-    session = createKeySpace(keyspace, secure = true, ssl = false)
+    session = createKeySpace(keyspace, secure = true)
     tableName = createTimestampTable(session, keyspace)
   }
 
@@ -54,33 +54,32 @@ class TestCassandraSourceTaskTimestamp extends WordSpec
 
   "A Cassandra SourceTask should read in incremental mode with timestamp and time slices" in {
     val taskContext = getSourceTaskContextDefault
-    val config = getCassandraConfigDefault()
+    val config = getCassandraConfigDefault
     val task = new CassandraSourceTask()
     task.initialize(taskContext)
 
     //start task
     task.start(config)
 
-    insertIntoTimestampTable(session, keyspace, tableName, "id1", "magic_string", getFormattedDateNow)
+    insertIntoTimestampTable(session, keyspace, tableName, "id1", "magic_string", getFormattedDateNow(), 1.toByte)
     
     var records = pollAndWait(task, tableName)
     var sourceRecord = records.asScala.head
     // check JSON fields
     var json: JsonNode = convertValueToJson(sourceRecord)
-    println(json)
     json.get("string_field").asText().equals("magic_string") shouldBe true
-    json.get("timestamp_field").asText().size > 0
+    json.get("timestamp_field").asText().nonEmpty shouldBe true
     json.get("int_field") shouldBe null
+    json.get("tinyint_field").asInt() shouldBe 1
 
-    insertIntoTimestampTable(session, keyspace, tableName, "id2", "magic_string2", getFormattedDateNow)
+    insertIntoTimestampTable(session, keyspace, tableName, "id2", "magic_string2", getFormattedDateNow(), 1.toByte)
 
     records = pollAndWait(task, tableName)
     sourceRecord = records.asScala.head
     // check JSON fields
     json = convertValueToJson(sourceRecord)
-    println(json)
     json.get("string_field").asText().equals("magic_string2") shouldBe true
-    json.get("timestamp_field").asText().size > 0
+    json.get("timestamp_field").asText().nonEmpty shouldBe true
     json.get("int_field") shouldBe null
 
     //stop task
@@ -97,7 +96,7 @@ class TestCassandraSourceTaskTimestamp extends WordSpec
     //start task
     task.start(config)
 
-    insertIntoTimestampTable(session, keyspace, tableName, "id1", "magic_string", getFormattedDateNow)
+    insertIntoTimestampTable(session, keyspace, tableName, "id1", "magic_string", getFormattedDateNow(), 1.toByte)
 
     val records = pollAndWait(task, tableName)
     val sourceRecord = records.asScala.head
@@ -113,7 +112,7 @@ class TestCassandraSourceTaskTimestamp extends WordSpec
   "A Cassandra SourceTask should read in incremental mode with fetchSize" in {
 
     val taskContext = getSourceTaskContextDefault
-    val config = getCassandraConfigDefault()
+    val config = getCassandraConfigDefault
     val task = new CassandraSourceTask()
 
     truncateTable(session, keyspace, tableName)
@@ -124,7 +123,7 @@ class TestCassandraSourceTaskTimestamp extends WordSpec
     task.start(config)
 
     for (i <- 1 to 10){
-      insertIntoTimestampTable(session, keyspace, tableName, s"id$i", s"magic_string_$i", getFormattedDateNow)
+      insertIntoTimestampTable(session, keyspace, tableName, s"id$i", s"magic_string_$i", getFormattedDateNow(), 1.toByte)
     }
 
     val records = pollAndWait(task, tableName)
@@ -142,7 +141,7 @@ class TestCassandraSourceTaskTimestamp extends WordSpec
     val task = new CassandraSourceTask()
     task.initialize(taskContext)
 
-    insertIntoTimestampTable(session, keyspace, tableName, "id1", "magic_string", getFormattedDateNow)
+    insertIntoTimestampTable(session, keyspace, tableName, "id1", "magic_string", getFormattedDateNow(), 1.toByte)
 
     try {
       task.start(config)
@@ -153,18 +152,18 @@ class TestCassandraSourceTaskTimestamp extends WordSpec
     task.stop()
   }
 
-  private def getCassandraConfigWithKcqlNoPrimaryKeyInSelect() = {
+  private def getCassandraConfigWithKcqlNoPrimaryKeyInSelect = {
     val myKcql = s"INSERT INTO sink_test SELECT string_field FROM $tableName PK timestamp_field INCREMENTALMODE=timestamp"
     getCassandraConfig(keyspace, tableName, myKcql)
   }
 
-  private def getCassandraConfigWithUnwrap() = {
+  private def getCassandraConfigWithUnwrap = {
     val myKcql = s"INSERT INTO sink_test SELECT string_field, timestamp_field FROM $tableName IGNORE timestamp_field PK timestamp_field WITHUNWRAP INCREMENTALMODE=timestamp"
     getCassandraConfig(keyspace, tableName, myKcql)
   }
 
-  private def getCassandraConfigDefault() = {
-    val myKcql = s"INSERT INTO sink_test SELECT string_field, timestamp_field FROM $tableName PK timestamp_field INCREMENTALMODE=timestamp"
+  private def getCassandraConfigDefault = {
+    val myKcql = s"INSERT INTO sink_test SELECT string_field, timestamp_field, tinyint_field FROM $tableName PK timestamp_field INCREMENTALMODE=timestamp"
     getCassandraConfig(keyspace, tableName, myKcql)
   }
 

--- a/kafka-connect-cassandra/src/test/scala/com/datamountaineer/streamreactor/connect/cassandra/source/TestCassandraSourceTaskTimestampLong.scala
+++ b/kafka-connect-cassandra/src/test/scala/com/datamountaineer/streamreactor/connect/cassandra/source/TestCassandraSourceTaskTimestampLong.scala
@@ -45,7 +45,7 @@ class TestCassandraSourceTaskTimestampLong extends WordSpec
   var tableName: String = _
 
   override def beforeAll {
-    session = createKeySpace(keyspace, secure = true, ssl = false)
+    session = createKeySpace(keyspace, secure = true)
     tableName = createTimestampTable(session, keyspace)
   }
 
@@ -64,7 +64,7 @@ class TestCassandraSourceTaskTimestampLong extends WordSpec
 
     val reader = CassandraTableReader(name = "test", session = session, setting = setting, context = taskContext, queue = queue)
 
-    insertIntoTimestampTable(session, keyspace, tableName, "id1", "magic_string", getFormattedDateNow)
+    insertIntoTimestampTable(session, keyspace, tableName, "id1", "magic_string", getFormattedDateNow(), 1.toByte)
 
     // clear out the default of Jan 1, 1900
     // and read the inserted row
@@ -82,14 +82,14 @@ class TestCassandraSourceTaskTimestampLong extends WordSpec
     json.get("string_field").asText().equals("magic_string") shouldBe true
 
     // insert another two records
-    insertIntoTimestampTable(session, keyspace, tableName, "id2", "magic_string2", getFormattedDateNow)
-    insertIntoTimestampTable(session, keyspace, tableName, "id3", "magic_string3", getFormattedDateNow)
+    insertIntoTimestampTable(session, keyspace, tableName, "id2", "magic_string2", getFormattedDateNow(), 1.toByte)
+    insertIntoTimestampTable(session, keyspace, tableName, "id3", "magic_string3", getFormattedDateNow(), 1.toByte)
 
     // sleep for longer than time slice (10 sec)
     Thread.sleep(11000)
 
     // insert another record
-    insertIntoTimestampTable(session, keyspace, tableName, "id4", "magic_string4", getFormattedDateNow)
+    insertIntoTimestampTable(session, keyspace, tableName, "id4", "magic_string4", getFormattedDateNow() , 1.toByte)
 
     //read
     reader.read()
@@ -126,7 +126,7 @@ class TestCassandraSourceTaskTimestampLong extends WordSpec
 
   }
 
-  private def getCassandraConfigDefault() = {
+  private def getCassandraConfigDefault = {
     val myKcql = s"INSERT INTO sink_test SELECT string_field, timestamp_field FROM $tableName PK timestamp_field INCREMENTALMODE=timestamp"
     getCassandraConfig(keyspace, tableName, myKcql)
   }

--- a/kafka-connect-cassandra/src/test/scala/com/datamountaineer/streamreactor/connect/cassandra/source/TestCassandraSourceTaskTimeuuid.scala
+++ b/kafka-connect-cassandra/src/test/scala/com/datamountaineer/streamreactor/connect/cassandra/source/TestCassandraSourceTaskTimeuuid.scala
@@ -16,10 +16,6 @@
 
 package com.datamountaineer.streamreactor.connect.cassandra.source
 
-import java.text.SimpleDateFormat
-import java.util.Date
-import java.util.concurrent.LinkedBlockingQueue
-
 import com.datamountaineer.streamreactor.connect.cassandra.TestConfig
 import com.datamountaineer.streamreactor.connect.schemas.ConverterUtil
 import com.fasterxml.jackson.databind.JsonNode
@@ -46,7 +42,7 @@ class TestCassandraSourceTaskTimeuuid extends WordSpec
   var tableName: String = _
 
   override def beforeAll {
-    session = createKeySpace(keyspace, secure = true, ssl = false)
+    session = createKeySpace(keyspace, secure = true)
     tableName = createTimeuuidTable(session, keyspace)
   }
 
@@ -70,9 +66,8 @@ class TestCassandraSourceTaskTimeuuid extends WordSpec
     var sourceRecord = records.asScala.head
     //check a field
     var json: JsonNode = convertValueToJson(sourceRecord)
-    println(json)
     json.get("string_field").asText().equals("magic_string") shouldBe true
-    json.get("timeuuid_field").asText().size > 0
+    json.get("timeuuid_field").asText().nonEmpty shouldBe true
     json.get("int_field") shouldBe null
 
 
@@ -82,9 +77,8 @@ class TestCassandraSourceTaskTimeuuid extends WordSpec
     sourceRecord = records.asScala.head
     //check a field
     json = convertValueToJson(sourceRecord)
-    println(json)
     json.get("string_field").asText().equals("magic_string2") shouldBe true
-    json.get("timeuuid_field").asText().size > 0
+    json.get("timeuuid_field").asText().nonEmpty shouldBe true
     json.get("int_field") shouldBe null
 
     //stop task
@@ -154,17 +148,17 @@ class TestCassandraSourceTaskTimeuuid extends WordSpec
     task.stop()
   }
 
-  private def getCassandraConfigWithKcqlNoPrimaryKeyInSelect() = {
+  private def getCassandraConfigWithKcqlNoPrimaryKeyInSelect = {
     val myKcql = s"INSERT INTO sink_test SELECT string_field FROM $tableName PK timeuuid_field INCREMENTALMODE=timeuuid"
     getCassandraConfig(keyspace, tableName, myKcql)
   }
 
-  private def getCassandraConfigWithUnwrap() = {
+  private def getCassandraConfigWithUnwrap = {
     val myKcql = s"INSERT INTO sink_test SELECT string_field, timeuuid_field FROM $tableName IGNORE timeuuid_field PK timeuuid_field WITHUNWRAP INCREMENTALMODE=timeuuid"
     getCassandraConfig(keyspace, tableName, myKcql)
   }
 
-  private def getCassandraConfigDefault() = {
+  private def getCassandraConfigDefault = {
     val myKcql = s"INSERT INTO sink_test SELECT string_field, timeuuid_field FROM $tableName PK timeuuid_field INCREMENTALMODE=timeuuid"
     getCassandraConfig(keyspace, tableName, myKcql)
   }

--- a/kafka-connect-cassandra/src/test/scala/com/datamountaineer/streamreactor/connect/cassandra/source/TestCassandraSourceTaskTimeuuidLong.scala
+++ b/kafka-connect-cassandra/src/test/scala/com/datamountaineer/streamreactor/connect/cassandra/source/TestCassandraSourceTaskTimeuuidLong.scala
@@ -45,7 +45,7 @@ class TestCassandraSourceTaskTimeuuidLong extends WordSpec
   var tableName: String = _
 
   override def beforeAll {
-    session = createKeySpace(keyspace, secure = true, ssl = false)
+    session = createKeySpace(keyspace, secure = true)
     tableName = createTimeuuidTable(session, keyspace)
   }
 
@@ -125,7 +125,7 @@ class TestCassandraSourceTaskTimeuuidLong extends WordSpec
 
   }
 
-  private def getCassandraConfigDefault() = {
+  private def getCassandraConfigDefault = {
     val myKcql = s"INSERT INTO sink_test SELECT string_field, timeuuid_field FROM $tableName PK timeuuid_field INCREMENTALMODE=timeuuid"
     getCassandraConfig(keyspace, tableName, myKcql)
   }

--- a/kafka-connect-cassandra/src/test/scala/com/datamountaineer/streamreactor/connect/cassandra/source/TestCassandraSourceUtil.scala
+++ b/kafka-connect-cassandra/src/test/scala/com/datamountaineer/streamreactor/connect/cassandra/source/TestCassandraSourceUtil.scala
@@ -22,17 +22,25 @@ trait TestCassandraSourceUtil {
         |long_field bigint,
         |string_field text, 
         |timestamp_field timestamp, 
-        |timeuuid_field timeuuid, 
+        |timeuuid_field timeuuid,
+        |tinyint_field tinyint,
         |PRIMARY KEY (id, timestamp_field)) WITH CLUSTERING ORDER BY (timestamp_field asc)""".stripMargin)
 
     table
   }
 
-  def insertIntoTimestampTable(session: Session, keyspace: String, tableName: String, anId: String, stringValue: String, formattedTimestamp: String) {
+  def insertIntoTimestampTable(session: Session,
+                               keyspace: String,
+                               tableName: String,
+                               anId: String,
+                               stringValue: String,
+                               formattedTimestamp: String,
+                               tinyint_field: Byte
+                              ) {
     val sql = s"""INSERT INTO $keyspace.$tableName
-      (id, int_field, long_field, string_field, timestamp_field, timeuuid_field)
+      (id, int_field, long_field, string_field, timestamp_field, timeuuid_field, tinyint_field)
       VALUES
-      ('$anId', 2, 3, '$stringValue', '$formattedTimestamp', now());"""
+      ('$anId', 2, 3, '$stringValue', '$formattedTimestamp', now(), $tinyint_field);"""
 
     // insert
     session.execute(sql)

